### ISSUE TITLE
Long Context

### ISSUE THREAD

**Issue by peregilk** [AI-Hypercomputer/maxtext#801]
I am trying to adapt Llama3 for long context, is 128k. I am training on a v5-256, and are trying to follow the procedure explained in https://arxiv.org/pdf/2407.14482. Basically this states:

````
 We set the batch size to 32 to fit 4 million tokens in a batch and use a learning rate of
3e-5 to train 2000 steps (8B tokens in total).
`````

I have prepared a dataset with 128k context, using the HF dataset. (Thanks @aireenmei). 

My challenge is however that setting ```per_device_batch_size=1``` gives me a global batch size of 256. This is way too high, and I get OOM errors. I want to split batches across devices. I attempted setting ```num_pipeline_microbatches=8```but this does not seem to work. 

Are there other ways of accomplishing this? I understand gradient accumulation is not implemented (and I am not sure if it will work here).

**Comment by aireenmei** [comment]
Have you tried setting per_device_batch_size<1 ?
See this example in our unit test: https://github.com/google/maxtext/blob/main/.github/workflows/UnitTests.yml#L102-L103

**Comment by peregilk** [comment]
Thanks. I did not notice this. I will test.

**Comment by peregilk** [comment]
@aireenmei. Took a while for me to be able to test this, since my tpus was busy with other stuff. Yes, it works! I struggled a to be able to get the mesh correct, but these are the settings that I think will allow me to train llama3 on a v5-256 with a bs=32 as described in the article above. 

```
max_target_length=131072
ici_fsdp_transpose_parallelism=1
ici_fsdp_parallelism=32
ici_tensor_parallelism=8
per_device_batch_size=0.25 

```
In addition I set `rope_max_timescale=150000000` in the config-file. There might be easier ways of doing this, but this was the only way I got it to work.

Thanks a lot for the help.

**Comment by peregilk** [comment]
@aireenmei Took a while for me to be able to test this, since my tpu's were busy with other stuff. Yes, it works! I struggled a bit to be able to get the mesh correct, but these are the settings that I think will allow me to train llama3 on a v5-256 with a global batch size of 32 as described in the article above. This is as close as I got:

```
max_target_length=65536
ici_fsdp_transpose_parallelism=1 
ici_fsdp_parallelism=32 
ici_tensor_parallelism=8
per_device_batch_size=0.125
allow_split_physical_axes=true
```

I had to reduce the target length slightly. I also needed to set the `allow_split_physical_axes=true`. In addition I set `rope_max_timescale=150000000` in the config-file. 

There might be better ways of doing this, and even increasing the target_length, but this was what I was able to get working. Maybe it can help someone else.

Thanks a lot for the help.
